### ISSUE TITLE
[Bug]: Calling decodeRecord on ChannelAction delete produces NPE

### ISSUE THREAD

**Issue by brezinajn** [jan-tennert/SupaCompose#1]
### General Info

- [X] I installed the latest version of SupaCompose
- [X] I check for similar bug report

### What happened? (include your code)

Calling decodeRecord on ChannelAction delete produced NPE

The reason is the !! operator.

``` Kotlin
inline fun <reified T> ChannelAction.decodeRecord(json: Json = Json) = json.decodeFromJsonElement<T>(record!!)
```

The problem stems from `record` being defined in `ChannelAction` interface so it forces all inheritors to implement it. Probably better solution would be to split this functionality to a `HasRecord` interface, implement it only for valid `ChannelAction`s and project `decodeRecord` over `HasRecord` instead of `ChannelAction`

I can tackle this one. Do you accept PRs?

### Platform

Android

### Relevant log output

```shell
E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-1
    Process: com.example.supabasetest, PID: 13722
    java.lang.NullPointerException
        at com.example.supabasetest.MainActivity$onCreate$1$1$1$1$invokeSuspend$lambda-4$$inlined$on$3.onEvent(RealtimeChannelBuilder.kt:65)
        at io.github.jan.supacompose.realtime.RealtimeChannelImpl.onMessage(RealtimeChannel.kt:88)
        at io.github.jan.supacompose.realtime.RealtimeImpl.onMessage(Realtime.kt:219)
        at io.github.jan.supacompose.realtime.RealtimeImpl.access$onMessage(Realtime.kt:108)
        at io.github.jan.supacompose.realtime.RealtimeImpl$listenForMessages$1.invokeSuspend(Realtime.kt:182)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
    	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@ad55b75, Dispatchers.Default]
D/RealtimeImpl$onMessage: Received event DELETE for channel realtime:public:Points
I/Process: Sending signal. PID: 13722 SIG: 9
```

**Comment by jan-tennert** [comment]
sure you can do a PR if you want :)

**Comment by jan-tennert** [comment]
Will be included in 0.1.2
